### PR TITLE
Release/4.0.0

### PR DESCRIPTION
**Pull request by bricesanchez**
We are closed to be able to release 4.0.0

* [x] Let's finish https://github.com/refinery/refinerycms-authentication-devise/pull/34 first
* [x] Release `refinerycms-authentication-devise` gem
* [x] Merge #3301 
* [x] Remove deprecated methods
* [x] Fix `rails console` start with dummy app

### CODE REVIEW

**Comment by bricesanchez** [comment]
@parndt Could you release `refinerycms-authentication-devise` ? Then we will be able to release 4.0.0

**Comment by parndt** [comment]
@bricesanchez I've released the authentication-devise package version 2.0.0

**Comment by bricesanchez** [comment]
@parndt could you release 4.0.0 ? 😄

File: changelog.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
-## 4.0.0.dev [unreleased]
+## 4.0.0 [unreleased]
 
-* [See full list](https://github.com/refinery/refinerycms/compare/3-0-stable...master)
+* Now support Rails 5.1. [#3122](https://github.com/refinery/refinerycms/pull/3122). [Brice Sanchez](https://github.com/bricesanchez) & [Philip Arndt](https://github.com/parndt) & [Remco Meinen](https://github.com/sjoulbak) & [Don Pinkster](https://github.com/donpinkster) & [Johan Bruning](https://github.com/johanb) & [Michael de Silva](https://github.com/bsodmike) & [Kiril Mitov](https://github.com/thebravoman) and all those we forgot!
```
Comment (parndt) [inline changelog.md]: I think you mean `Now supporting` or `Now supports` 😄

Comment (parndt) [reply changelog.md]: Instead of `all those we forgot` could you put something like `everyone else who helped out`? 👍

File: readme.md [hunk 2]
```diff
@@ -30,9 +30,13 @@ If you're new to Refinery, start with this guide:
 
 * __[Getting Started](http://refinerycms.com/guides/getting-started)__
 
-If you want to use `3-0-stable` branch of Refinery with Rails 4.2.x. Install using this template:
+If you want to use `4.0.x` branch of Refinery with Rails 5.1.x. Install using this template:
```
Comment (parndt) [inline readme.md]: This branch will be called `4-0-stable`

Comment (parndt) [reply readme.md]: But I think you mean to say 
>If you want to use version `4.0.x` of Refinery CMS which supports Rails 5.1+

File: readme.md [hunk 3]
```diff
@@ -30,9 +30,13 @@ If you're new to Refinery, start with this guide:
...
 
-    rails new app_name -m http://refinerycms.com/t/3-0-stable
+    rails new app_name -m http://refinerycms.com/t/4.0.0
+
+If you want to use `3.0.x` branch of Refinery with Rails 4.2.x. Install using this template:
```
Comment (parndt) [inline readme.md]: this should say `3-0-stable` branch

**Comment by bricesanchez** [comment]
@parndt i've made all changes.

**Comment by bricesanchez** [comment]
🍿 😄

**Comment by parndt** [comment]
@bricesanchez I get a heap of warnings about our version requirements in the gemspec files when I build the gems, so I wanted to look at that first.

**Comment by bricesanchez** [comment]
Is c0ee713 could help you?

**Comment by bricesanchez** [comment]
@parndt I've seen that we didn't clean up deprecated methods expected for 3.1 version

**Comment by parndt** [comment]
Good catch!

**Comment by bricesanchez** [comment]
What's the next step?

**Comment by bricesanchez** [comment]
There is maybe a bug: 

```
➜  refinerycms git:(release/4.0.0) ✗ rc
/Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/railties-5.1.4/lib/rails/command/actions.rb:12:in `require': cannot load such file -- /Users/bricesanchez/Sites/refinery/refinerycms (LoadError)
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/railties-5.1.4/lib/rails/command/actions.rb:12:in `require_application_and_environment!'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/railties-5.1.4/lib/rails/commands/console/console_command.rb:96:in `perform'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/railties-5.1.4/lib/rails/command/base.rb:63:in `perform'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/railties-5.1.4/lib/rails/command.rb:44:in `invoke'
	from /Users/bricesanchez/Sites/refinery/refinerycms/.gems/gems/railties-5.1.4/lib/rails/commands.rb:16:in `<top (required)>'
	from /Users/bricesanchez/Sites/refinery/refinerycms/spec/dummy/bin/rails:4:in `require'
	from /Users/bricesanchez/Sites/refinery/refinerycms/spec/dummy/bin/rails:4:in `<top (required)>'
	from bin/rails:5:in `load'
	from bin/rails:5:in `<main>'
```

**Comment by parndt** [comment]
What are the steps to reproduce it?

**Comment by bricesanchez** [comment]
> What are the steps to reproduce it?

1. `git checkout release/4.0.0`
2. `bundle update`
3. `rake refinery:testing:dummy_app`
4. `rails console`

then you should see this error:

```
/Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.4/lib/rails/command/actions.rb:12:in `require': cannot load such file -- /Users/bricesanchez/Sites/refinery/refinerycms (LoadError)
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.4/lib/rails/command/actions.rb:12:in `require_application_and_environment!'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.4/lib/rails/commands/console/console_command.rb:96:in `perform'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.4/lib/rails/command/base.rb:63:in `perform'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.4/lib/rails/command.rb:44:in `invoke'
	from /Users/bricesanchez/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/railties-5.1.4/lib/rails/commands.rb:16:in `<top (required)>'
	from /Users/bricesanchez/Sites/refinery/refinerycms/spec/dummy/bin/rails:4:in `require'
	from /Users/bricesanchez/Sites/refinery/refinerycms/spec/dummy/bin/rails:4:in `<top (required)>'
	from bin/rails:5:in `load'
	from bin/rails:5:in `<main>'
```

**Comment by bricesanchez** [comment]
@parndt , the bug is present since Rails 5.1.0

In https://github.com/refinery/refinerycms/blob/master/bin/rails#L4
if i change line 4 for : 

```
ENGINE_PATH = File.expand_path('../../lib/refinery/all',  __FILE__)
```

I'm able to start the console in the dummy_app.

But i don't if it's the best way to do... what do you think?

**Comment by parndt** [comment]
@bricesanchez nice find - with your help I was able to create ef0c228 which I think nicely solves the problem?

**Comment by bricesanchez** [comment]
@parndt i confirm, your commit fix the problem :)

**Comment by bricesanchez** [comment]
Could i merge it @parndt ?

**Comment by bricesanchez** [comment]
Thanks @parndt ! Are you ok to release the gems too?

**Comment by parndt** [comment]
Yeah, I'll address this soon. 😄

**Comment by parndt** [comment]
Done!

**Comment by bricesanchez** [comment]
Thank you @parndt !

**Pull request merged**